modules/runner/src/main/java/org/apache/ignite/internal/configuration/ClusterConfigurationDefaultsSetter.java

```diff
@@ -0,0 +1,146 @@
+
+    private static final String DEFAULT_USERNAME = "ignite";
+
+    private static final String DEFAULT_PASSWORD = "ignite";
+
+    private final ConfigurationManager configurationManager;
+
+    private final ConfigurationValidatorImpl configurationValidator;
+
+    /**
+     * Constructor.
+     *
+     * @param rootKeys Root keys.
+     * @param generator Configuration tree generator.
+     */
+    public ClusterConfigurationDefaultsSetter(
+            Collection<RootKey<?, ?>> rootKeys,
+            ConfigurationTreeGenerator generator
+    ) {
+        configurationValidator = new ConfigurationValidatorImpl(generator, Set.of());
+        configurationManager = new ConfigurationManager(
+                rootKeys,
+                new RuntimeConfigurationStorage(ConfigurationType.DISTRIBUTED),
+                generator,
+                configurationValidator
+        );
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public CompletableFuture<String> setDefaults(String hocon) {
+        ConverterToMapVisitor visitor = ConverterToMapVisitor.builder()
+                .includeInternal(true)
+                .maskSecretValues(false)
+                .build();
+
+        ConfigurationRegistry configurationRegistry = configurationManager.configurationRegistry();
+
+        return validateCfg(hocon)
+                .thenCompose(v -> parseHocon(hocon))
+                .thenCompose(
+                        v -> setSecurityDefaults(configurationRegistry.getConfiguration(SecurityConfiguration.KEY))
+                )
+                .thenApply(
+                        v -> HoconConverter.represent(configurationRegistry, List.of(), visitor).render(ConfigRenderOptions.concise())
+                );
+    }
+
+    private CompletableFuture<Void> validateCfg(String hocon) {
+        try {
+            List<ValidationIssue> validationIssues = configurationValidator.validateHocon(hocon);
+            if (validationIssues.isEmpty()) {
+                return CompletableFuture.completedFuture(null);
+            } else {
+                return failedFuture(new ConfigurationValidationException(validationIssues));
+            }
+        } catch (Exception e) {
+            return failedFuture(new ConfigurationValidationException(e.getMessage()));
+        }
+    }
+
+    private CompletableFuture<Void> parseHocon(String hocon) {
+        try {
+            Config config = ConfigFactory.parseString(hocon);
+            ConfigurationSource hoconSource = HoconConverter.hoconSource(config.root());
+            ConfigurationRegistry configurationRegistry = configurationManager.configurationRegistry();
+
+            return configurationRegistry.change(hoconSource);
+        } catch (Exception e) {
+            return failedFuture(e);
+        }
+    }
+
+    private static CompletableFuture<Void> setSecurityDefaults(SecurityConfiguration securityCfg) {
+        if (securityCfg.authentication().providers().value().size() == 0) {
```

Comment:
Instead of updating `*Configuration` instance, you could update `*View` instance. It's much simpler and doesn't require the storage.
Let's re-write this class a bit, I don't like current approach